llvm/lib/Target/LoongArch/LoongArchISelLowering.h

```diff
@@ -271,7 +271,7 @@ class LoongArchTargetLowering : public TargetLowering {
       unsigned *Fast = nullptr) const override;
 
   bool isShuffleMaskLegal(ArrayRef<int> Mask, EVT VT) const override {
-    return false;
+    return isTypeLegal(VT.getSimpleVT());
```

Comment:
It would be safer to add a check for `VT.isSimple()` before proceeding with `getSimpleVT()`.